### ISSUE TITLE
Still problems when using templates in NC Office with  NC AIO 10.1.1

### ISSUE THREAD

**Issue by 4tler** [nextcloud/all-in-one#5763]
@szaimen i testet #5688 it but the problem is still exist.

I still have 2 instances that have this problem. The strange thing is that the files are created after the error message, but not opened.

The instances are in an LXC via Proxmox.

I have already tried this without success: 

- Delete/reinstall NC Office completely
- Activate/deactivate antivirus

> 
> Message in NC LOG:
> 
> <details><summary>Details</summary>
> <p>
> 
> ```
> {
>   "reqId": "KwCoddSAVIKkefDPyhsN",
>   "level": 3,
>   "time": "2024-12-13T16:21:44+00:00",
>   "remoteAddr": "IP",
>   "user": "user",
>   "app": "no app in context",
>   "method": "POST",
>   "url": "/ocs/v2.php/apps/files/api/v1/templates/create",
>   "message": "Client error: `POST https://cloud.domain.de/cool/transform-document-structure` resulted in a `403 Forbidden` response",
>   "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36",
>   "version": "30.0.4.1",
>   "exception": {
>     "Exception": "GuzzleHttp\\Exception\\ClientException",
>     "Message": "Client error: `POST https://cloud.domain.de/cool/transform-document-structure` resulted in a `403 Forbidden` response",
>     "Code": 403,
>     "Trace": [
>       {
>         "file": "/var/www/html/3rdparty/guzzlehttp/guzzle/src/Middleware.php",
>         "line": 72,
>         "function": "create",
>         "class": "GuzzleHttp\\Exception\\RequestException",
>         "type": "::",
>         "args": [
>           "*** sensitive parameters replaced ***"
>         ]
>       },
>       {
>         "file": "/var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php",
>         "line": 209,
>         "function": "GuzzleHttp\\{closure}",
>         "class": "GuzzleHttp\\Middleware",
>         "type": "::",
>         "args": [
>           "*** sensitive parameters replaced ***"
>         ]
>       },
>       {
>         "file": "/var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php",
>         "line": 158,
>         "function": "callHandler",
>         "class": "GuzzleHttp\\Promise\\Promise",
>         "type": "::",
>         "args": [
>           1,
>           "*** sensitive parameters replaced ***",
>           "*** sensitive parameters replaced ***"
>         ]
>       },
>       {
>         "file": "/var/www/html/3rdparty/guzzlehttp/promises/src/TaskQueue.php",
>         "line": 52,
>         "function": "GuzzleHttp\\Promise\\{closure}",
>         "class": "GuzzleHttp\\Promise\\Promise",
>         "type": "::",
>         "args": [
>           "*** sensitive parameters replaced ***"
>         ]
>       },
>       {
>         "file": "/var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php",
>         "line": 251,
>         "function": "run",
>         "class": "GuzzleHttp\\Promise\\TaskQueue",
>         "type": "->",
>         "args": [
>           true
>         ]
>       },
>       {
>         "file": "/var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php",
>         "line": 227,
>         "function": "invokeWaitFn",
>         "class": "GuzzleHttp\\Promise\\Promise",
>         "type": "->",
>         "args": []
>       },
>       {
>         "file": "/var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php",
>         "line": 272,
>         "function": "waitIfPending",
>         "class": "GuzzleHttp\\Promise\\Promise",
>         "type": "->",
>         "args": []
>       },
>       {
>         "file": "/var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php",
>         "line": 229,
>         "function": "invokeWaitList",
>         "class": "GuzzleHttp\\Promise\\Promise",
>         "type": "->",
>         "args": []
>       },
>       {
>         "file": "/var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php",
>         "line": 69,
>         "function": "waitIfPending",
>         "class": "GuzzleHttp\\Promise\\Promise",
>         "type": "->",
>         "args": []
>       },
>       {
>         "file": "/var/www/html/3rdparty/guzzlehttp/guzzle/src/Client.php",
>         "line": 189,
>         "function": "wait",
>         "class": "GuzzleHttp\\Promise\\Promise",
>         "type": "->",
>         "args": []
>       },
>       {
>         "file": "/var/www/html/lib/private/Http/Client/Client.php",
>         "line": 276,
>         "function": "request",
>         "class": "GuzzleHttp\\Client",
>         "type": "->",
>         "args": [
>           "post",
>           "*** sensitive parameters replaced ***",
>           {
>             "0": "And 1 more entries, set log level to debug to see all entries",
>             "verify": "/var/www/html/resources/config/ca-bundle.crt",
>             "timeout": 5,
>             "nextcloud": {
>               "allow_local_address": true
>             },
>             "multipart": [
>               {
>                 "name": "data",
>                 "contents": null,
>                 "headers": {
>                   "Content-Type": "multipart/form-data"
>                 }
>               },
>               {
>                 "name": "transform",
>                 "contents": "{\"Transforms\": []}"
>               },
>               {
>                 "name": "format",
>                 "contents": "ott"
>               }
>             ],
>             "headers": {
>               "User-Agent": "Nextcloud Server Crawler",
>               "Accept-Encoding": "gzip"
>             }
>           }
>         ]
>       },
>       {
>         "file": "/var/www/html/custom_apps/richdocuments/lib/Service/TemplateFieldService.php",
>         "line": 186,
>         "function": "post",
>         "class": "OC\\Http\\Client\\Client",
>         "type": "->",
>         "args": [
>           "*** sensitive parameters replaced ***"
>         ]
>       },
>       {
>         "file": "/var/www/html/custom_apps/richdocuments/lib/Listener/FileCreatedFromTemplateListener.php",
>         "line": 61,
>         "function": "fillFields",
>         "class": "OCA\\Richdocuments\\Service\\TemplateFieldService",
>         "type": "->",
>         "args": [
>           {
>             "__class__": "OC\\Files\\Node\\File"
>           },
>           "*** sensitive parameters replaced ***"
>         ]
>       },
>       {
>         "file": "/var/www/html/lib/private/EventDispatcher/ServiceEventListener.php",
>         "line": 68,
>         "function": "handle",
>         "class": "OCA\\Richdocuments\\Listener\\FileCreatedFromTemplateListener",
>         "type": "->",
>         "args": [
>           {
>             "__class__": "OCP\\Files\\Template\\FileCreatedFromTemplateEvent"
>           }
>         ]
>       },
>       {
>         "file": "/var/www/html/3rdparty/symfony/event-dispatcher/EventDispatcher.php",
>         "line": 220,
>         "function": "__invoke",
>         "class": "OC\\EventDispatcher\\ServiceEventListener",
>         "type": "->",
>         "args": [
>           {
>             "__class__": "OCP\\Files\\Template\\FileCreatedFromTemplateEvent"
>           },
>           "OCP\\Files\\Template\\FileCreatedFromTemplateEvent",
>           {
>             "__class__": "Symfony\\Component\\EventDispatcher\\EventDispatcher"
>           }
>         ]
>       },
>       {
>         "file": "/var/www/html/3rdparty/symfony/event-dispatcher/EventDispatcher.php",
>         "line": 56,
>         "function": "callListeners",
>         "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
>         "type": "->",
>         "args": [
>           [
>             {
>               "__class__": "Closure"
>             }
>           ],
>           "OCP\\Files\\Template\\FileCreatedFromTemplateEvent",
>           {
>             "__class__": "OCP\\Files\\Template\\FileCreatedFromTemplateEvent"
>           }
>         ]
>       },
>       {
>         "file": "/var/www/html/lib/private/EventDispatcher/EventDispatcher.php",
>         "line": 67,
>         "function": "dispatch",
>         "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
>         "type": "->",
>         "args": [
>           {
>             "__class__": "OCP\\Files\\Template\\FileCreatedFromTemplateEvent"
>           },
>           "OCP\\Files\\Template\\FileCreatedFromTemplateEvent"
>         ]
>       },
>       {
>         "file": "/var/www/html/lib/private/EventDispatcher/EventDispatcher.php",
>         "line": 79,
>         "function": "dispatch",
>         "class": "OC\\EventDispatcher\\EventDispatcher",
>         "type": "->",
>         "args": [
>           "OCP\\Files\\Template\\FileCreatedFromTemplateEvent",
>           {
>             "__class__": "OCP\\Files\\Template\\FileCreatedFromTemplateEvent"
>           }
>         ]
>       },
>       {
>         "file": "/var/www/html/lib/private/Files/Template/TemplateManager.php",
>         "line": 162,
>         "function": "dispatchTyped",
>         "class": "OC\\EventDispatcher\\EventDispatcher",
>         "type": "->",
>         "args": [
>           {
>             "__class__": "OCP\\Files\\Template\\FileCreatedFromTemplateEvent"
>           }
>         ]
>       },
>       {
>         "file": "/var/www/html/apps/files/lib/Controller/TemplateController.php",
>         "line": 68,
>         "function": "createFromTemplate",
>         "class": "OC\\Files\\Template\\TemplateManager",
>         "type": "->",
>         "args": [
>           "*** sensitive parameters replaced ***"
>         ]
>       },
>       {
>         "file": "/var/www/html/lib/private/AppFramework/Http/Dispatcher.php",
>         "line": 208,
>         "function": "create",
>         "class": "OCA\\Files\\Controller\\TemplateController",
>         "type": "->",
>         "args": [
>           "*** sensitive parameters replaced ***"
>         ]
>       },
>       {
>         "file": "/var/www/html/lib/private/AppFramework/Http/Dispatcher.php",
>         "line": 114,
>         "function": "executeController",
>         "class": "OC\\AppFramework\\Http\\Dispatcher",
>         "type": "->",
>         "args": [
>           {
>             "__class__": "OCA\\Files\\Controller\\TemplateController"
>           },
>           "create"
>         ]
>       },
>       {
>         "file": "/var/www/html/lib/private/AppFramework/App.php",
>         "line": 161,
>         "function": "dispatch",
>         "class": "OC\\AppFramework\\Http\\Dispatcher",
>         "type": "->",
>         "args": [
>           {
>             "__class__": "OCA\\Files\\Controller\\TemplateController"
>           },
>           "create"
>         ]
>       },
>       {
>         "file": "/var/www/html/lib/private/Route/Router.php",
>         "line": 302,
>         "function": "main",
>         "class": "OC\\AppFramework\\App",
>         "type": "::",
>         "args": [
>           "OCA\\Files\\Controller\\TemplateController",
>           "create",
>           {
>             "__class__": "OC\\AppFramework\\DependencyInjection\\DIContainer"
>           },
>           {
>             "_route": "ocs.files.template.create"
>           }
>         ]
>       },
>       {
>         "file": "/var/www/html/ocs/v1.php",
>         "line": 43,
>         "function": "match",
>         "class": "OC\\Route\\Router",
>         "type": "->",
>         "args": [
>           "/ocsapp/apps/files/api/v1/templates/create"
>         ]
>       },
>       {
>         "file": "/var/www/html/ocs/v2.php",
>         "line": 7,
>         "args": [
>           "/var/www/html/ocs/v1.php"
>         ],
>         "function": "require_once"
>       }
>     ],
>     "File": "/var/www/html/3rdparty/guzzlehttp/guzzle/src/Exception/RequestException.php",
>     "Line": 113,
>     "message": "Client error: `POST https://cloud.domain.de/cool/transform-document-structure` resulted in a `403 Forbidden` response",
>     "exception": [],
>     "CustomMessage": "Client error: `POST https://cloud.domain.de/cool/transform-document-structure` resulted in a `403 Forbidden` response"
>   },
>   "id": "675c5fecb6660"
> ``` 
> 
> </p>
> </details> 
> 
> <details><summary>Details</summary>
> <p>
> 
> ```
> {
>   "reqId": "KwCoddSAVIKkefDPyhsN",
>   "level": 3,
>   "time": "2024-12-13T16:21:44+00:00",
>   "remoteAddr": "IP",
>   "user": "USER",
>   "app": "richdocuments",
>   "method": "POST",
>   "url": "/ocs/v2.php/apps/files/api/v1/templates/create",
>   "message": "Client error: `POST https://cloud.domain/cool/transform-document-structure` resulted in a `403 Forbidden` response",
>   "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36",
>   "version": "30.0.4.1",
>   "data": {
>     "app": "richdocuments"
>   },
>   "id": "675c5fecb6667"
> }
> ``` 
> 
> </p>
> </details> 
> 
> ![Image](https://github.com/user-attachments/assets/fe0ab429-7482-4155-95b1-4744019619af)
>  

 _Originally posted by @4tler in [#5682](https://github.com/nextcloud/all-in-one/issues/5682#issuecomment-2541819775)_

**Comment by szaimen** [comment]
Hi, can you please post the collabora contianer logs here? There should be a log message that will help us to further debug this issue.

**Comment by szaimen** [comment]
Okay, I can reproduce this:
```2024-12-17T10:04:02.027667553Z wsd-00006-00028 2024-12-17 11:04:02.024722 +0100 [ websrv_poll ] WRN  convert-to: Requesting address is denied: fdf2:24b8:2e1c::1| wsd/ClientRequestDispatcher.cpp:443
2024-12-17T10:04:02.027748983Z wsd-00006-00028 2024-12-17 11:04:02.026589 +0100 [ websrv_poll ] WRN  #25: Conversion requests not allowed from this address: ::ffff:172.18.0.11| wsd/ClientRequestDispatcher.cpp:1512
```
I tried to make it work with adding `--o:net.post_allow_host[20]=0\\.0\\.0\\.0/0 --o:net.post_allow_host[21]=::/0 --o:net.post_allow_host[22]=::ffff:172\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3} --o:net.post_allow_host[23]=::ffff:172\\.18\\.0\\.11 --o:net.post_allow_host[24]=fdf2:24b8:2e1c::1` but did not succeed yet. Will need to ask internally for help on this as I am out of ideas now.

**Comment by szaimen** [comment]
Okay, found a fix in https://github.com/nextcloud/all-in-one/pull/5781

**Comment by szaimen** [comment]
This is now fixed with v10.2.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by 4tler** [comment]
@szaimen it works with my test instance. Big thanks!
